app/models/authentication.rb

```diff
@@ -82,9 +82,10 @@ def self.new(*args, &block)
   RETRYABLE_STATUS = %w[error unreachable].freeze
 
   CREDENTIAL_TYPES = {
-    :external_credential_types         => 'ManageIQ::Providers::ExternalAutomationManager::Authentication',
-    :embedded_ansible_credential_types => 'ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Credential',
-    :workflows_credential_types        => 'ManageIQ::Providers::Workflows::AutomationManager::Credential'
+    :external_credential_types           => 'ManageIQ::Providers::ExternalAutomationManager::Authentication',
+    :embedded_ansible_credential_types   => 'ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Credential',
+    :workflows_credential_types          => 'ManageIQ::Providers::Workflows::AutomationManager::Credential',
+    :embedded_terraform_credential_types => 'ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential'
```

Comment:
@agrare Perhaps not for this PR, but is it possible to make this pluggable?  I know it's for the UI, but I'm not sure exactly in what capacity.

Offhand, if feel like this could just be a method in Vmdb::Plugins somehow.

Reply:
I noticed there's "External" types and then we enumerate the Embedded types, so another options is it could collapse down to the EmbeddedAutomationManager .

Reply:
It looks like this is only used for `OPTIONS /api/authentications` I'm not sure why we do this as opposed to the way we do every other OPTIONS method with DDF params but it might be to group the available types, so e.g. `Workflows::AutomationManager::ScmCredential` and `Workflows::AutomationManager::WorkflowCredential` under `EmbeddedWorkflows`

We could definitely do this in a manner that doesn't require a hard-coded list in the base class.

Reply:
Made pluggable with https://github.com/ManageIQ/manageiq/pull/22995

Reply:
@GilbertCherrie @nasark has resolved this, can you rebase on master?

Reply:
Oh actually that was the only change here so I think this can be closed :tada: